specs/add_utils_to_common.adoc

```diff
@@ -0,0 +1,163 @@
+= Add general purpose utils to common
+
+
+Common repository should have utilities for few very frequently needed actions
+like command execution, package installation, service management etc.
+
+== Problem description
+
+There are few very frequently executed actions like like command execution,
+package installation, service management etc. in the tendrl backend. It is
+better to have utility for these in tendrl-common, so that all the backend
+components like tendrl-node-agent tendrl-gluster-integration and tendrl-ceph
+integration can perform those actions by using tendrl-common as library. This
+avoids the duplication of effort needed to implement such actions. And makes
+maintainability easier
+
+== Use Cases
+
+* All tendrl-backand components can use tendrl-common as library for performing
+these actions.
+
+== Proposed change
+
+Add commonly needed tendrl-backend actions/functions as utils to tendrl-common.
+So that tendrl-common can be used as a library to perform such actions. The
+commonly used actions that we see in tendrl backend currently are:
+
+* command-execution
+* package-installation
+* service-management
+
+All these functions are available as ansible modules. So if we can write a
+wrapper to utilize ansible modules, we will get the functionality without
+having to actually implement the functionality. This avoids a lot of dev
+effort. Also these modules are maintained by ansible community which is very
+active and these modules should be well tested as its being used by a huge
+community. And ansible has a very good collection of modules, so in future
+if we find out that a new util is needed then most probably an ansible
+module will be already available for that. We just have to use the module
+through the wrapper. So its better to leverage the ansible module for
+adding utils to tendrl-common
+
+The expected usage of utils is as follows:
+
+[source,python]
+----
+# using command utility
+
+from tendrl.common.utils.command import Command
+command = Command("ls -la") # it takes command as argument
```

Comment:
yes the command-executor will be in common. Have included this info in the latest commit. Please take a look.